### PR TITLE
Add comparison (equality and inequality) operators to Grids.

### PR DESCRIPTION
**Pull request by ehpor**
This allows grids to be compared for equality:

```python
if grid1 == grid2:
    print('These two grids are equal.')
else:
    print('These two grids are different.')
```

The inequality operator (`!=`) is defined implicitly by Python 3 based on the equality operator.

Note that even if Grids contain the same coordinates but are different types, then the comparison will return False. This is intentional.

### CODE REVIEW

File: tests/test_field.py [hunk 1]
```diff
@@ -188,33 +188,53 @@ def test_field_svd():
...
 
 	grid2 = CartesianGrid(SeparatedCoords(copy.deepcopy(grid1.separated_coords)))
 	assert hash(grid1) != hash(grid2)
+	assert grid1 != grid2
+	assert grid2 != grid1
 
 	grid3 = CartesianGrid(UnstructuredCoords(copy.deepcopy(grid1.coords)))
 	assert hash(grid1) != hash(grid3)
+	assert grid1 != grid3
+	assert grid3 != grid1
```
Comment (syhaffert) [inline tests/test_field.py:203]: Add an assert grid2 != grid3. Otherwise we will miss a code check for unstructure vs separated coords.

**Review by syhaffert — CHANGES_REQUESTED** [review]
I have 1 small comment.

File: tests/test_field.py (hunk 1 above)
Comment (ehpor) [reply tests/test_field.py:205]: Done. Also added the reverse (which goes through a different code path.

**Review by syhaffert — APPROVED** [review]

**Comment by ehpor** [comment]
Force merging because of the last commit (due to merges right before this one); the review is still valid.

**Pull request merged**